dio/lib/src/dio_exception.dart

```diff
@@ -132,14 +132,26 @@ class DioException implements Exception {
   }) =>
       DioException(
         type: DioExceptionType.receiveTimeout,
+        requestOptions: requestOptions,
+        response: null,
+        error: error,
         message: 'The request took longer than $timeout to receive data. '
             'It was aborted. '
             'To get rid of this exception, try raising the '
             'RequestOptions.receiveTimeout above the duration of $timeout or '
             'improve the response time of the server.',
+      );
+
+  factory DioException.badCertificate({
+    required RequestOptions requestOptions,
+    Object? error,
+  }) =>
+      DioException(
+        type: DioExceptionType.badCertificate,
         requestOptions: requestOptions,
         response: null,
         error: error,
+        message: 'The certificate of the response is not approved.',
```

Comment:
Isn't it just an "invalid" certificate where invalid is defined either by the app or the environment the app is running in? "approved" seems somewhat misleading if that's the case

Reply:
Per,

https://github.com/cfug/dio/blob/1c2843a2fab1c3795272df1e81502616761e192e/dio/lib/src/adapters/io_adapter.dart#L51-L57

https://github.com/cfug/dio/blob/1c2843a2fab1c3795272df1e81502616761e192e/dio/lib/src/adapters/io_adapter.dart#L183-L199

The message means that the user does not approve the certificate of the response by themselves.

Reply:
Got it, thanks!